collector/exporter/otelarrowexporter/internal/arrow/stream.go

```diff
@@ -340,6 +340,14 @@ func (s *Stream) encodeAndSend(wri writeItem, hdrsBuf *bytes.Buffer, hdrsEnc *hp
 	ctx, span := s.tracer.Start(wri.parent, "otel_arrow_stream_send")
 	defer span.End()
 
+	var err error
+	defer func() {
+		// Due to potential double compression the netstats code knows uncompressed bytes
+		// value can be unreliable. Add span attributes for uncompressed size and set
+		// span Status if an error is returned.
+		s.netReporter.SetSpanSizeAttributes(ctx, sized)
+		s.netReporter.SetSpanError(ctx, err)
+	}
```

Comment:
I think it would be clearer if the call to `SetSpanSizeAttributes` was alongside the call to `s.netReporter.CountSend` below, so no extra defer is needed. The span error can be inside the defer here, but (as commented above) can be written directly, no need for netstats to set span status).

Reply:
Fixed